features/step_definitions/basic_steps.rb

```diff
@@ -14,40 +14,9 @@
   fill_in field, with: text
 end
 
-Given('We have the following user:') do |table|
-  table.hashes.each do |field|
-    create(:user, field)
-  end
-end
-
-Given('We have the following recipes:') do |table|
-  table.hashes.each do |recipe|
-    if recipe[:user]
-      recipe[:user] = User.find_by email: recipe[:user]
-    end
-
-    if recipe["forked from"]
-      recipe[:original_recipe_id] = Recipe.find_by(title: recipe["forked from"]).id if recipe["forked from"] != ""
-      recipe.delete("forked from")
-    end
-
-    if recipe[:category]
-      recipe[:category] = Category.find_by name: recipe[:category]
-    end
-
-    new_recipe = create(:recipe, recipe.except('image'))
-
-    if recipe[:image]
-      new_recipe.image.attach(io: File.open("#{::Rails.root}/spec/fixtures/#{recipe[:image]}"),
-                              filename: "#{recipe[:image]}",
-                              content_type: "image/png")
-    end
-  end
-end
-
 Given("I am logged in as {string}") do |user_email|
-  @user = User.find_by(email: user_email)
-  login_as @user
+  @user =  user = User.find_by(email: user_email) || create(:user, email: user_email)
```

Comment:
???

Reply:
?????? clarify please. I have removed the `user =`if that was what you meant i dont know why this was like that.

Reply:
Yes, that’s what I meant